src/js/utils/generateCache.js

```diff
@@ -69,6 +69,9 @@ export default function generateCache() {
         } else if (Object.prototype.toString.call(video.thumbnail) === '[object String]') {
           assetsToCache.push(video.thumbnail);
         }
+        if (Array.isArray(video['media-session-artwork'])) {
```

Comment:
Is this needed to cache media session artwork images if the Media Session API is not supported?

Reply:
@beaufortfrancois I was wondering if we even want to cache those files. It's quite a lot of data – 400 kB currently – and the benefit of the artwork being available in the offline mode is marginal IMO. We can always fall back to the default artwork in those cases.

And if you're asking whether we should exclude browsers with no Media Session API support, then I'd say _perhaps_. With Safari 15 on iOS out the door, there is currently [no mainstream evergreen browser that would _not_ support it](https://caniuse.com/?search=mediasession).

![image](https://user-images.githubusercontent.com/433570/135499186-d952ef52-71b1-4061-b4ba-8bc07d7c4e87.png)

What is your opinion on the caching?

Reply:
> What is your opinion on the caching?

Is it possible to cache them only when media session metadata are set?
In other words, only when the media starts playing.

Reply:
I think adding logic to the SW to cache this artwork when it's played, or dowloaded for offline playback, would be a better user experience instead of caching all the artwork during initialization.

Reply:
Good idea, guys, thank you. I've implemented the changes in d7ac82071faf6134c9a019640bcc6e3491b38361

The artwork is now cached when a video is played or downloaded.

Reply:
QQ: Is this called caching assets every time video is played and/or downloaded or only once?

Reply:
@beaufortfrancois Aha, good point, I did not realize `cache.addAll` would overwrite the cache entries. Updated the `saveToCache` implementation in 64ca1720018b2b8555b6533f463f2086ec0d4fa9. Could you please take a look?

I believe it is sufficient to `match` the URL, because we're versioning cache. LMK if you think this needs to be more robust. Thanks!